docs/ref/settings.txt

```diff
@@ -264,9 +264,10 @@ The cache connection to use for the :ref:`cache middleware
 
 Default: ``''`` (Empty string)
 
-A string which will be prefixed to the cache keys generated by the :ref:`cache
-middleware <the-per-site-cache>`. This prefix is combined with the
-:setting:`KEY_PREFIX <CACHES-KEY_PREFIX>` setting; it does not replace it.
+A string - or a callable that returns a string - which will be prefixed to the
```

Comment:
This should accept a string containing a dotted path to a callable, rather than a callable itself.

Also, I don't think that `CACHE_MIDDLEWARE_KEY_PREFIX` can accept a callable in the current version of this patch (see `django.utils.cache`).

Reply:
Oh, I missed `django.utils.cache`. I think I can update it to check whether `key_prefix` is a callable and act accordingly - I see `request` _is_ available.

I'm not sure I understand why the key prefix should be a dotted path to a callable as opposed to a callable.

Reply:
> Oh, I missed `django.utils.cache`. I think I can update it to check whether `key_prefix` is a callable and act accordingly - I see `request` _is_ available.
> 
> I'm not sure I understand why the key prefix should be a dotted path to a callable as opposed to a callable.

This would be more flexible, consistent with other settings, and would also help avoiding circular imports when a callable is defined outside of settings files.

Reply:
I don't think the Django decision process should factor other people's circular import errors. The dotted path pattern works well as a form of DI to e.g. configure the SMTP backend via env vars. It protects you from circular import errors, but not from import errors, and I presume this is only accidental.

Here the key prefix callable is intrinsically dynamic. Making it dynamically configurable seems superfluous i.e. dynamic key prefix callable for a dynamic key prefix?

Besides, how do you tell the difference between "foo.bar.baz", the key prefix, and "foo.bar.baz", the key prefix callable?